### PR TITLE
Status: Add Visitor class

### PR DESCRIPTION
**Pull request by kraftbj**
Discussed in #22269

I added it to Status, but like @jeherve, I'm open to a better place for it.

#### Changes proposed in this Pull Request:

* See above.

#### Jetpack product discussion

n/a

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* n/a
* Does the location make sense?

### CODE REVIEW

**Review by fgiannar — COMMENTED** [review]
Thanks for working on this @kraftbj !

I took this opportunity to add some unit tests, hope that's ok :)

I could see this class in the **tracking** package as well, populated in the future with various visitor related info, such as: device (device-detection package), geo, etc

**Comment by kraftbj** [comment]
❤️ I love unit tests. Thank you @fgiannar!

That's a good point. The tracking package doesn't feel great since this isn't for the a8c Tracks system or a8c tracking. I don't know that Status is a slam dunk either, but thinking things related to the Visitor's status (e.g. IP, location, mobile device), etc could make sense?

File: projects/plugins/jetpack/_inc/lib/plans.php [hunk 1]
```diff
@@ -36,7 +36,7 @@ public static function get_plans() {
 			array(
 				'method'  => 'GET',
 				'headers' => array(
-					'X-Forwarded-For' => Jetpack::current_user_ip( true ),
+					'X-Forwarded-For' => (new Automattic\Jetpack\Status\Visitor())->get_ip( true ),
```
Comment (jeherve) [inline projects/plugins/jetpack/_inc/lib/plans.php:39]:
```suggestion
					'X-Forwarded-For' => ( new Automattic\Jetpack\Status\Visitor() )->get_ip( true ),
```

**Review by jeherve — CHANGES_REQUESTED** [review]
> The tracking package doesn't feel great since this isn't for the a8c Tracks system or a8c tracking. I don't know that Status is a slam dunk either, but thinking things related to the Visitor's status (e.g. IP, location, mobile device), etc could make sense?

Maybe we would need a brand new package just for those things? Something like Visitor, as you named the class, or server-info?

*******

Could you run `tools/fixup-project-versions.sh` once again to update version numbers?

File: projects/plugins/jetpack/_inc/lib/plans.php (hunk 1 above)
Comment (kraftbj) [reply projects/plugins/jetpack/_inc/lib/plans.php:39]: 1d7748564d

**Comment by kraftbj** [comment]
> Maybe we would need a brand new package just for those things? Something like Visitor, as you named the class, or server-info?

Do we see a situation where we'd ever want that _without_ the rest of the Status package?

I'll let it sit for the weekend.

**Comment by fgiannar** [comment]
> The tracking package doesn't feel great since this isn't for the a8c Tracks system or a8c tracking. I don't know that Status is a slam dunk either, but thinking things related to the Visitor's status (e.g. IP, location, mobile device), etc could make sense?

I mostly suggested the tracking package because usually contexts such as 'Visitor' or 'Visit' are related to tracking. My intuitive reaction, for example, would be to look for such functionality under the `tracking` package, however this might be just me :)

A brand new `Visitor` package, as Jeremy suggests, would also make sense to me, however we might want to avoid the overhead of creating a new package right away and leave this in `Status` for now until we decide.

**Comment by jeherve** [comment]
> however we might want to avoid the overhead of creating a new package right away and leave this in Status for now until we decide.

Let's go with the new Class in the Status package then. 🚢

**Review by jeherve — DISMISSED** [review]

**Review by oskosk — APPROVED** [review]

**Comment by anomiex** [comment]
Don't forget to take care of D73018-code!

**Comment by kraftbj** [comment]
Thanks for the reminder! I'll handle this early in my day tomorrow since it requires the package to be added to WP.com and not just a quick deploy :-D.

**Comment by jeherve** [comment]
r239663-wpcom

**Pull request merged**